src/osd/PrimaryLogPG.cc

```diff
@@ -11439,7 +11439,9 @@ bool PrimaryLogPG::primary_error(
        ++i) {
     if (*i == get_primary()) continue;
     pg_shard_t peer = *i;
-    if (!peer_missing[peer].is_missing(soid, v)) {
+    pg_missing_item mi;
+    bool missing = peer_missing[peer].is_missing(soid, &mi);
+    if (!missing || mi.have == v) {
```

Comment:
I would rename the bool _missing_ to _got_mi_ to avoid confusion.  Also, shouldn't the test be something like this?

`if (!missing || mi.have != v) {`

Reply:
> if (!missing || mi.have != v) {

That is not true.
If the peer does not miss this object or is missing this object but still have the version that the primary currently wants,
we can add this peer to missing_loc and use it for later recovery.